staging_vespalib/src/vespa/vespalib/util/singleexecutor.h

```diff
@@ -39,12 +40,21 @@ class SingleExecutor final : public vespalib::SyncableThreadExecutor, vespalib::
     void drain_tasks();
     void sleepProducer(Lock & guard, duration maxWaitTime, uint64_t wakeupAt);
     void run_tasks_till(uint64_t available);
-    void wait_for_room(Lock & guard);
+    Task::UP wait_for_room_or_put_in_overflow_Q(Lock & guard, Task::UP task);
+    uint64_t move_to_main_q(Lock & guard, Task::UP task);
+    void move_overflow_to_main_q();
+    void move_overflow_to_main_q(Lock & guard);
     uint64_t index(uint64_t counter) const {
         return counter & (_taskLimit.load(std::memory_order_relaxed) - 1);
     }
 
     uint64_t numTasks() const {
+        return num_task_in_main_q() + num_tasks_in_overflow_q();
```

Comment:
this is no longer thread safe to call from the executor thread due to overflow->size()

Reply:
This has been split up into different version were guard is taken when not already held.